src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -51,7 +52,11 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 context.SyntaxProvider
 
                 // Find all MethodDeclarationSyntax nodes attributed with RegexGenerator and gather the required information.
-                .CreateSyntaxProvider(IsSyntaxTargetForGeneration, GetSemanticTargetForGeneration)
+                .ForAttributeWithMetadataName(
+                    context,
+                    RegexGeneratorAttributeName,
+                    (n, c) => n is MethodDeclarationSyntax,
+                    GetSemanticTargetForGeneration)
```

Comment:
in the future you can delete all the code i've added and this line will now work against the *actual* ForAttributeWithMetadataName extension we expose.  teh only difference is you will also delete the `context,` argument.  That argument is not necessary with our extension because the extension has special access to that data as a private field in the SyntaxProvider type.

Reply:
Lovely